### PR TITLE
Opsexp 651: Automate version update of deployment components in the readme

### PR DESCRIPTION
**Pull request by Mieszko96**
Scenario
Someone made changes in values
Someone want to commit chase
Pre-commit will fail (Readme updated by script) ﻿
Someone need to “git add .” ﻿
Someone commit again(with automated updated Readme file)

Tested, if new file 8.0.N-extra-vars.yml added to repository, will be included into table.

Problems:
developers using this repository need to have installed python3 with pyyaml library.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -34,3 +34,10 @@ repos:
     hooks:
       - id: giticket
         args: ['--regex', '[A-Z]+-\d+[0-9]', '--format', '{ticket}: {commit_msg}', '--mode', 'regex_match']
+
+  - repo: local
+    hooks:
+      - id: custom
```
Comment (alxgomz) [inline .pre-commit-config.yaml:40]: I would use a more descriptive id is possible

File: scripts/generate-comp-ver-table.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            line_in_table+=f"{version} | "
+
+        line_in_table=line_in_table[:-1]
+        end_table+=f"{line_in_table}\n"
+    return end_table
+
+def modify_table():
+    """Delete old table from README.md, and put new one"""
+    delete_next_lines = False
+    a_file = open("docs/README.md", "r", encoding="utf-8")
```
Comment (alxgomz) [inline scripts/generate-comp-ver-table.py:112]: Does this pass pylint? I thought with syntax was now the way to go with python3

File: scripts/generate-comp-ver-table.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        if "| Sync Service | " in line.strip("\n"):
+            delete_next_lines = False
+            new_file.write(get_content_of_new_table())
+        elif delete_next_lines is True:
+            pass
+        elif "| Component | " in line.strip("\n"):
+            delete_next_lines = True
+        else:
+            new_file.write(line)
+
```
Comment (alxgomz) [inline scripts/generate-comp-ver-table.py:128]: That logic relies heavily on the fact the table will always keep the same structure, ordering, labelling and so on...
I'm afraid this makes it somewhat fragile.
Did you investigate whether a markdown library would have allowed table manipulation in a more solid manner? (Just wondering, I have not done any investigation myself)

**Review by alxgomz — CHANGES_REQUESTED** [review]
Only the later comment is a real concern

File: scripts/generate-comp-ver-table.py (hunk 3 above)
Comment (Mieszko96) [reply scripts/generate-comp-ver-table.py:128]: I will make investigation

File: scripts/generate-comp-ver-table.py (hunk 2 above)
Comment (Mieszko96) [reply scripts/generate-comp-ver-table.py:112]: Passed pylint, i can change to "with open"

File: .pre-commit-config.yaml (hunk 1 above)
Comment (Mieszko96) [reply .pre-commit-config.yaml:40]: Will change

File: scripts/generate-comp-ver-table.py (hunk 3 above)
Comment (Mieszko96) [reply scripts/generate-comp-ver-table.py:128]: One idea is to create this table in seperate file. In README.md will be reference to this file.

Comment (alxgomz) [reply scripts/generate-comp-ver-table.py]: as agreed with @mteodori it's ok to keep it like that

**Review by alxgomz — APPROVED** [review]
LGTM

**Review by alxgomz — CHANGES_REQUESTED** [review]
Can you please add pyyaml to the `requirements.txt` file and add that check to the Linters stage (in case dev doesn't have pre-commit installed locally?

**Comment by Mieszko96** [comment]
> Can you please add pyyaml to the `requirements.txt` file and add that check to the Linters stage (in case dev doesn't have pre-commit installed locally?

File: .travis.yml [hunk 4]
```diff
@@ -39,6 +39,11 @@ jobs:
     - name: "yamllint"
       script:
         - yamllint .
+    - name: Check components version table
+      cripts:
```
Comment (alxgomz) [inline .travis.yml:43]: looks like a typo

File: scripts/generate-comp-ver-table.py [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+"""Generate components version table in docs/README.md based on extra-vars.yml."""
+
+import sys
+import glob
+import yaml
+
+
+def get_acs_extra_values_files():
+    """Return list of extra-values files in repository."""
```
Comment (alxgomz) [inline scripts/generate-comp-ver-table.py:11]: you still have an extra-values instead of extra-vars here

**Review by alxgomz — APPROVED** [review]
LGTM

**Pull request merged**